PGM/src/main/java/tc/oc/pgm/mutation/types/kit/TeamChestMutation.java

```diff
@@ -0,0 +1,80 @@
+package tc.oc.pgm.mutation.types.kit;
+
+import net.md_5.bungee.api.ChatColor;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.PlayerInventory;
+import tc.oc.commons.bukkit.chat.ComponentRenderContext;
+import tc.oc.commons.bukkit.item.RenderedItemBuilder;
+import tc.oc.commons.core.chat.Component;
+import tc.oc.pgm.kits.FreeItemKit;
+import tc.oc.pgm.kits.Kit;
+import tc.oc.pgm.match.Match;
+import tc.oc.pgm.match.MatchPlayer;
+import tc.oc.pgm.mutation.Mutation;
+import tc.oc.pgm.mutation.types.KitMutation;
+import tc.oc.pgm.teams.Team;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+public class TeamChestMutation extends KitMutation {
+    private String title = Mutation.TEAMCHEST.getName();
+
+    private Map<Team, Inventory> teamInventories = new WeakHashMap<>();
+
+    @Inject private static RenderedItemBuilder.Factory itemBuilders;
+    @Inject private static ComponentRenderContext componentRenderContext;
```

Comment:
I'm pretty sure static injections shouldn't be here, not completly sure where you should get them from though, mutations seem not to be very injectable via constructor, as they seem to be built from just the match, probably should be injected into the match scope by the main mutation class, instead of just created from this constructor